File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(count, sum):
        nonlocal answer
        
        if count == len(numbers):
            if sum == target:
                answer += 1
            return

        dfs(count + 1, sum + numbers[count])
        dfs(count + 1, sum - numbers[count])
    
    dfs(0, 0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    static int answer = 0;
    
    public int solution(int[] numbers, int target) {
        dfs(0, numbers, target, 0);
        
        return answer;
    }
    
    public void dfs(int currentNumber, int[] numbers, int target, int count) {
        if (count == numbers.length) {
            if (currentNumber == target) {
                answer++;
            }
            return;
        }
        
        dfs(currentNumber + numbers[count], numbers, target, count + 1);
        dfs(currentNumber - numbers[count], numbers, target, count + 1);
    }
}